### ISSUE TITLE
Make timeout parameters consensus parameters

### ISSUE THREAD

**Issue by williambanfield** [tendermint/tendermint#7275]
Currently, our timeout parameters exist in the validator-local config. [Link](https://github.com/tendermint/tendermint/blob/f28d629e280ddcdc0dd644ccf1586d73dddfb7a1/config/config.go#L922)

However, these parameters must be the same across all validators so it does not make sense for them to be part of the local config file. This issue proposes removing the parameters from the local configuration file and placing them into the consensus parameters.

Related to: https://github.com/tendermint/tendermint/issues/7274
Duplicates: https://github.com/tendermint/tendermint/issues/2920

**Comment by ebuchman** [comment]
> However, these parameters must be the same across all validators

I don't think this is actually true? cc @milosevic @josef-widder aren't these local parameters? No can actually verify what timeouts are being used so even if we make these consensus based parameters, node operators could still in principle change them.

The paper does simplify the paper set, as outlined in https://github.com/tendermint/tendermint/issues/7274. So maybe the thing that should be in governance is the `∆`, which is technically supposed to be an unknown value, but in principle is a network wide value and I guess we could put that in consensus and have all the other parameters be functions of that?

**Comment by williambanfield** [comment]
Thanks for taking a look @ebuchman

> However, these parameters must be the same across all validators

Ah, this is an imprecision in the way that I stated this. What I mean is that for a network to function correctly, these parameters must be the same across all validators. These are currently local parameters. Meaning, node operators have an easy ability to update these parameters by editing their own local configuration. If the parameters were drastically different across validators, then I don't believe you would maintain liveness guarantees since validators could begin to proceed through rounds at very different rates.

I've written [ADR-74](https://github.com/tendermint/tendermint/pull/7503) to capture the steps needed to change the local parameters into consensus parameters with some justification for using a similar set of timeout parameters as we have now. To summarize the reasoning for keeping more than two values for these timeouts: gossiping a proposed block should take much more time than gossiping a vote. If we used just two parameters, we'd likely have to set the vote timeouts to match the proposal timeout and this would cause slowdown in some cases.
